src/shapes/text.class.ts

```diff
@@ -1,1821 +1,1793 @@
+   * @default
+   */
+  pathStartOffset: number;
+
+  /**
+   * Which side of the path the text should be drawn on.
+   * Only used when text has a path
+   * @type {String} 'left|right'
+   * @default
+   */
+  pathSide: string;
+
+  /**
+   * How text is aligned to the path. This property determines
+   * the perpendicular position of each character relative to the path.
+   * (one of "baseline", "center", "ascender", "descender")
+   * This feature is in BETA, and its behavior may change
+   * @type String
+   * @default
+   */
+  pathAlign: string;
+
+  /**
+   * @private
+   */
+  _fontSizeFraction: number;
+
+  /**
+   * @private
+   */
+  offsets: { underline: number; linethrough: number; overline: number };
+
+  /**
+   * Text Line proportion to font Size (in pixels)
+   * @type Number
+   * @default
+   */
+  _fontSizeMult: number;
+
+  /**
+   * additional space between characters
+   * expressed in thousands of em unit
+   * @type Number
+   * @default
+   */
+  charSpacing: number;
+
+  /**
+   * Reference to a context to measure text char or couple of chars
+   * the cacheContext of the canvas will be used or a freshly created one if the object is not on canvas
+   * once created it will be referenced on fabric._measuringContext to avoid creating a canvas for every
+   * text object created.
+   * @type {CanvasRenderingContext2D}
+   * @default
+   */
+  _measuringContext: CanvasRenderingContext2D | null = null;
+
+  /**
+   * Baseline shift, styles only, keep at 0 for the main text object
+   * @type {Number}
+   * @default
+   */
+  deltaY: number;
+
+  /**
+   * WARNING: EXPERIMENTAL. NOT SUPPORTED YET
+   * determine the direction of the text.
+   * This has to be set manually together with textAlign and originX for proper
+   * experience.
+   * some interesting link for the future
+   * https://www.w3.org/International/questions/qa-bidi-unicode-controls
+   * @since 4.5.0
+   * @type {String} 'ltr|rtl'
+   * @default
+   */
+  direction: string;
+
+  /**
+   * Array of properties that define a style unit (of 'styles').
+   * @type {Array}
+   * @default
+   */
+  _styleProperties: (keyof this)[];
+
+  /**
+   * contains characters bounding boxes
+   */
+  protected __charBounds: {
+    left: number;
+    width: number;
+    kernedWidth: number;
+    height: number;
+  }[] = [];
+
+  /**
+   * use this size when measuring text. To avoid IE11 rounding errors
+   * @type {Number}
+   * @default
+   * @readonly
+   * @private
+   */
+  CACHE_FONT_SIZE: number;
+
+  /**
+   * contains the min text width to avoid getting 0
+   * @type {Number}
+   * @default
+   */
+  MIN_TEXT_WIDTH: number;
+
+  protected __skipDimension: boolean;
+  protected textLines: string[];
+  protected _textLines: string[][];
+  protected _unwrappedTextLines: string[][];
+  protected _text: string[];
+  protected cursorWidth: number;
+  protected __lineHeights: number[];
+  protected __lineWidths: number[];
+  protected _forceClearCache: boolean;
```

Comment:
some of those are useful to read in certain cases. I'm not sure we should flag them as protected as part of the migration

Reply:
I can remove. I have no opinion on this

Reply:
i m about to commit some comments in the source code and a forgot var/const let's not overlap. i m about to merge this